pyluna-pathology/luna/pathology/cli/generate_tiles.py

```diff
@@ -106,41 +108,39 @@ def generate_tiles(input_slide_image, tile_size, requested_magnification, output
 
     df = pd.DataFrame(address_raster).set_index("address")
 
-    output_binary_file = f"{output_dir}/{slide_name}.tiles.pil"
+    output_hdf_file = f"{output_dir}/{slide_name}.tiles.h5"
     output_header_file = f"{output_dir}/{slide_name}.tiles.csv"
 
-    fp = open(output_binary_file, 'wb')
-    offset = 0
-    counter = 0
     logger.info(f"Now generating tiles with num_cores={num_cores} and batch_size={batch_size}!")
 
-    address_offset = []
+    address_tiles = []
     with ProcessPoolExecutor(num_cores) as executor:
-        out = [executor.submit(get_tile_bytes, index, input_slide_image, full_resolution_tile_size, tile_size ) for index in grouper(df.index, batch_size)]
+        out = [executor.submit(get_tile_arrays, index, input_slide_image, full_resolution_tile_size, tile_size )
+               for index in grouper(df.index, batch_size)]
         for future in tqdm(as_completed(out), file=sys.stdout, total=len(out)):
-            for index, tile in future.result(): 
-                fp.write( tile )
-                
-                address_offset.append ((index, int(offset)))
-                offset += len(tile)
-                counter+=1
-    fp.close()
+            for index, tile in future.result():
+                address_tiles.append({index:tile})
```

Comment:
We'd want to call `hfile.create_dataset(index, data=tile)` here so we don't have to save all tiles in memory, and can spend the time as the futures compute to do the disk IO